otherlibs/unix/accept_win32.c

```diff
@@ -40,7 +44,24 @@ CAMLprim value caml_unix_accept(value cloexec, value sock)
     caml_win32_maperr(err);
     caml_uerror("accept", Nothing);
   }
-  caml_win32_set_cloexec((HANDLE) snew, cloexec);
+
+  /* accept sockets should be non-inheritable (cloexec) by default
+   * but let's check anyway */
+  if (! GetHandleInformation((HANDLE) snew, &flags)) {
+    caml_win32_maperr(GetLastError());
+    caml_uerror("accept", Nothing);
+  }
+  /* duplicate the socket if the requested inheritance bit doesn't
+   * match the socket */
+  if (!!(flags & HANDLE_FLAG_INHERIT) != inherit) {
```

Comment:
This is a bit too clever for my taste, but don't have anything simpler to propose.